vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher.cc

```diff
@@ -795,5 +792,104 @@ void Publisher::OnServerPublisherInfoLoaded(
   callback(std::move(server_info));
 }
 
+void Publisher::UpdateMediaDuration(
+    const uint64_t window_id,
+    const std::string& publisher_key,
+    uint64_t duration) {
+  BLOG(1, "Media duration: " << duration);
+  ledger_->database()->GetPublisherInfo(publisher_key,
+      std::bind(&Publisher::OnGetPublisherInfoForUpdateMediaDuration,
+                this,
+                _1,
+                _2,
+                window_id,
+                duration));
+}
+
+void Publisher::OnGetPublisherInfoForUpdateMediaDuration(
+    type::Result result,
+    type::PublisherInfoPtr info,
+    const uint64_t window_id,
+    uint64_t duration) {
+  if (result != type::Result::LEDGER_OK) {
+    BLOG(0, "Failed to retrieve publisher info while updating media duration");
+    return;
+  }
+
+  type::VisitData visit_data;
+  visit_data.name = info->name;
+  visit_data.url = info->url;
+  visit_data.provider = info->provider;
+  visit_data.favicon_url = info->favicon_url;
+
+  SaveVideoVisit(
+      info->id,
+      visit_data,
+      duration,
+      window_id,
+      [=](type::Result, type::PublisherInfoPtr) {});
```

Comment:
Probably best to have an empty capture clause here.

Reply:
Fixed.